packages/nextjs/src/shared/deprecated.ts

```diff
@@ -0,0 +1,63 @@
+// copy of packages/shared/src/utils/deprecated.ts
```

Comment:
We need to spend time on fixing this error and the copy/paste should only be the very last resort.

What happens if you put `"use client"` into the `usePagesOrInfinite` file?

Reply:
I wanted to avoid introducing changes in `@clerk/shared` that would require extensive testing in this PR so I followed the naive approach of copy / paste to proceed with this PR objective and introduce another PR to support importing `@clerk/shared` in server-side frameworks.
Based on your comment, if we consider blocking this PR until the @clerk/shared is fixed and used I have opened the following PR in which we should also check if the @clerk/shared is tree-shakeable or not before we merge it: https://github.com/clerkinc/javascript/pull/1767
cc: @SokratisVidros